modern/src/UpdateCheckPrompt.tsx

```diff
@@ -0,0 +1,57 @@
+import { Snackbar, Button } from '@mui/material';
+import React from 'react'
+import { useTranslation } from './common/components/LocalizationProvider';
+import { useAttributePreference } from './common/util/preferences';
+import { useRegisterSW } from 'virtual:pwa-register/react'
+
+// Based on https://vite-pwa-org.netlify.app/frameworks/react.html
+function UpdateCheckPrompt() {
+  const t = useTranslation();
+
+  const serviceWorkerUpdateInterval = useAttributePreference('serviceWorkerUpdateInterval', 3600000);
+
+  const {
+    needRefresh: [needRefresh],
+    updateServiceWorker,
+  } = useRegisterSW({
+    onRegisteredSW(swUrl, swRegistration) {
+      if (serviceWorkerUpdateInterval > 0 && swRegistration) {
+        setInterval(async () => {
+          if (!(!swRegistration.installing && navigator)) {
+            return;
+          }
+    
+          if (('connection' in navigator) && !navigator.onLine) {
+            return;
+          }
+    
+          const newSW = await fetch(swUrl, {
+            cache: 'no-store',
+            headers: {
+              'cache': 'no-store',
+              'cache-control': 'no-cache',
+            },
+          });
+    
+          if (newSW?.status === 200) {
+            await swRegistration.update();
+          }
+        }, serviceWorkerUpdateInterval);
+      }
+    }
+  });
+
+  return (
+    <Snackbar 
+      open={needRefresh} 
+      message={t('settingsUpdateAvailable')}
+      action={(
+        <Button onClick={() => updateServiceWorker(true)}>
+          {t('settingsReload')}
+        </Button>
```

Comment:
Can we just use an icon button? Like this one:

https://mui.com/material-ui/material-icons/?query=ref&selected=Refresh

It looks better and also we can remove one of the strings.

Reply:
Good idea, did not even think about this.
![image](https://github.com/traccar/traccar-web/assets/858821/9b9511e3-adf2-468a-8cc9-f3356f00de8e)